gitlabbot/gitlabbot/handler.go

```diff
@@ -139,53 +130,48 @@ func (h *Handler) handleSubscribeToBranch(cmd string, msg chat1.MsgSummary, crea
 		return nil
 	}
 	args := toks[2:]
-	var message string
-	defer func() {
-		if message != "" {
-			_, err = h.kbc.SendMessageByConvID(msg.ConvID, fmt.Sprintf(message, args[0], args[1]))
-			if err != nil {
-				err = fmt.Errorf("error sending message: %s", err)
-			}
-		}
-	}()
-
 	if len(args) < 2 {
 		return fmt.Errorf("bad args for subscribe to branch: %v", args)
 	}
 
 	defaultBranch := "master"
-
-	if exists, err := h.db.GetSubscriptionExists(msg.ConvID, args[0], defaultBranch); !exists {
-		if err != nil {
-			return fmt.Errorf("error getting subscription: %s", err)
-		}
-		var message string
+	repo := args[0]
+	branch := args[1]
+	exists, err := h.db.GetSubscriptionExists(msg.ConvID, repo, defaultBranch)
+	if err != nil {
+		return err
+	} else if !exists {
 		if create {
-			message = fmt.Sprintf("You aren't subscribed to updates yet!\nSend this first: `!github subscribe %s`", args[0])
+			_, err = h.kbc.SendMessageByTlfName(msg.Sender.Username, formatSetupInstructions(repo, msg, h.httpPrefix, h.secret))
+			if err != nil {
+				return fmt.Errorf("error sending message: %s", err)
+			}
+			if !base.IsDirectPrivateMessage(msg) {
+				h.ChatEcho(msg.ConvID, "OK! I've sent a message to @%s to authorize me.", msg.Sender.Username)
+			}
 		} else {
-			message = fmt.Sprintf("You aren't subscribed to notifications for %s!", args[0])
-		}
-		_, err := h.kbc.SendMessageByConvID(msg.ConvID, message)
-		if err != nil {
-			return fmt.Errorf("Error sending message: %s", err)
+			h.ChatEcho(msg.ConvID, "You aren't subscribed to notifications for `%s`!", repo)
```

Comment:
So ChatEcho goes to wherever the user initiated the message?

Reply:
Just echos the given message to the given convID https://github.com/keybase/managed-bots/blob/master/base/output.go#L66-L70

I prefer it since it masks the error handling internally and makes hte bot logic easier to read